packages/kiota_http/lib/src/http_client_request_adapter.dart

```diff
@@ -0,0 +1,272 @@
+part of '../kiota_http.dart';
+
+class HttpClientRequestAdapter implements RequestAdapter {
+  HttpClientRequestAdapter({
+    required http.Client client,
+    required AuthenticationProvider authProvider,
+    required ParseNodeFactory pNodeFactory,
+    required SerializationWriterFactory sWriterFactory,
+  })  : _client = client,
+        _authProvider = authProvider,
+        _pNodeFactory = pNodeFactory,
+        _sWriterFactory = sWriterFactory;
+
+  static const String _claimsKey = 'claims';
+
+  final http.Client _client;
+  final AuthenticationProvider _authProvider;
+  ParseNodeFactory _pNodeFactory;
+  SerializationWriterFactory _sWriterFactory;
+  String? _baseUrl;
+
+  @override
+  String? get baseUrl => _baseUrl;
+
+  @override
+  set baseUrl(String? value) {
+    if (value?.endsWith('/') ?? false) {
+      _baseUrl = value?.substring(0, value.length - 1);
+    } else {
+      _baseUrl = value;
+    }
+  }
+
+  Future<http.BaseRequest> _getMessageFromInfo(
+    RequestInformation requestInfo,
+  ) async {
+    final request = http.Request(requestInfo.httpMethod!.name, requestInfo.uri);
+
+    // TODO(ricardoboss): implement the rest of the method
```

Comment:
This means at the moment requests with bodies/headers can't be sent. yes?

Reply:
Whoops! Yeah, I forgot to implement that...